src/rviz/ogre_helpers/stl_loader.cpp

```diff
@@ -91,6 +91,13 @@ bool STLLoader::load(const std::string& path)
 bool STLLoader::load(uint8_t* buffer)
 {
   uint8_t* pos = buffer;
+
+  // check for ascii type
+  if(std::string((char*)buffer,5) == std::string("solid")) {
+    ROS_ERROR("Trying to load STL file from binary buffer, but the given data is an ASCII STL file. Please convert it to a binary STL file instead.");
```

Comment:
Another nitpick: This line is really long, can you please do something like this:

```c++
ROS_ERROR("Trying to load STL file from binary buffer, but the given data is an ASCII STL file. "
          "Please convert it to a binary STL file instead.");
```